compress_test.go

```diff
@@ -0,0 +1,124 @@
+// Go MySQL Driver - A MySQL-Driver for Go's database/sql package
+//
+// Copyright 2024 The Go-MySQL-Driver Authors. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package mysql
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"testing"
+)
+
+func makeRandByteSlice(size int) []byte {
+	randBytes := make([]byte, size)
+	rand.Read(randBytes)
+	return randBytes
+}
+
+// compressHelper compresses uncompressedPacket and checks state variables
+func compressHelper(t *testing.T, mc *mysqlConn, uncompressedPacket []byte) []byte {
+	conn := new(mockConn)
+	mc.netConn = conn
+
+	n, err := mc.writeCompressed(uncompressedPacket)
```

Comment:
Since writeCompressed does not call mc.syncSequenceNr(), do you need to add it here? I tried it locally and that makes the TestRoundtrip pass.